com.unity.render-pipelines.universal/Editor/Lighting/UniversalRenderPipelineLightUI.Drawers.cs

```diff
@@ -112,6 +112,16 @@ static Func<int> SetGizmosDirty()
         }
 
         static void DrawGeneralContent(UniversalRenderPipelineSerializedLight serializedLight, Editor owner)
+        {
+            DrawGeneralContentInternal(serializedLight, owner, isInPreset: false);
+        }
+
+        internal static void DrawGeneralContentPreset(UniversalRenderPipelineSerializedLight serializedLight, Editor owner)
+        {
+            DrawGeneralContentInternal(serializedLight, owner, isInPreset: true);
+        }
+
+        static void DrawGeneralContentInternal(UniversalRenderPipelineSerializedLight serializedLight, Editor owner, bool isInPreset)
```

Comment:
Personally I would only keep one `DrawGeneralContent` and take in an argument whether Preset mode is active. `DrawGeneralContent` and `DrawGeneralContentInternal` have the same accessibility level, so it does not enforce an usage. Furthermore, if more options or modes are added in the future the number of methods would expand a lot. That being said this is internal code and will be easy to refactor in the future.

Reply:
Yes i do agree with you. The problem is that we are building the UI with the CoreEditorDrawer APIs which expects a specific signature so i cannot just add the parameter "IsPreset".

Reply:
Could it be done with a ConditionalDrawer?